core/commonMain/src/ArgType.kt

```diff
@@ -70,17 +70,43 @@ abstract class ArgType<T : Any>(val hasParameter: kotlin.Boolean) {
                     ?: throw ParsingException("Option $name is expected to be double number. $value is provided.")
     }
 
+    companion object {
+        /**
+         * Helper for arguments that have limited set of possible values represented as enumeration constants.
+         */
+        inline fun <reified T: Enum<T>> Choice(
+            noinline toVariant: (kotlin.String) -> T = { enumValueOf(it.toUpperCase()) },
```

Comment:
This assumes that enum variants are named in upper case. I'd rather make case-insensitive search for enums by default.